java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -178,190 +232,129 @@ final boolean isSuperOrThisMemberSelect(ExpressionTree tree) {
       return false;
     }
 
-  }
-
-  public class ConditionVisitor extends BaseExpressionVisitor {
-    ExecutionState currentState;
-    PackedStates currentResult;
-
-    public PackedStates evaluate(ExecutionState state, ExpressionTree tree) {
-      currentState = state;
-      currentResult = new PackedStates();
-      scan(tree);
-      return currentResult;
-    }
-
     @Override
     public final void visitArrayAccessExpression(ArrayAccessExpressionTree tree) {
-      evaluateExpression(currentState, tree.expression());
-      evaluateExpression(currentState, tree.index());
-      currentResult.unknownStates.add(currentState);
-    }
-
-    @Override
-    public final void visitAssignmentExpression(AssignmentExpressionTree tree) {
-      evaluateExpression(currentState, tree.variable());
-      evaluateExpression(currentState, tree.expression());
-      Symbol.VariableSymbol symbol = extractVariableSymbol(tree.variable());
-      if (symbol != null) {
-        currentState.setBooleanConstraint(new SymbolicValue.SymbolicVariableValue(symbol), SymbolicBooleanConstraint.UNKNOWN);
+      super.visitArrayAccessExpression(tree);
+      for (ExecutionState executionState : currentStates) {
+        executionState.pop();
+        executionState.pop();
+        executionState.push(new SymbolicValue());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this instantiation of "SymbolicValue". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2440)